src/main/java/org/folio/service/inventory/InventoryInstanceManager.java

```diff
@@ -347,17 +343,47 @@ private Future<String> createInstance(JsonObject instanceRecJson, RequestContext
   public Future<CompositePoLine> openOrderHandleInstance(CompositePoLine compPOL, boolean isInstanceMatchingDisabled, RequestContext requestContext) {
     return consortiumConfigurationService.getConsortiumConfiguration(requestContext)
       .compose(consortiumConfiguration -> {
-        if (compPOL.getInstanceId() != null) {
-          return consortiumConfiguration.isEmpty() ? Future.succeededFuture(compPOL.getInstanceId())
-            : shareInstanceAmongTenantsIfNeeded(compPOL.getInstanceId(), consortiumConfiguration.get(), compPOL.getLocations(), requestContext);
-        } else {
-          return getInstanceRecord(compPOL, isInstanceMatchingDisabled, requestContext)
-            .compose(instanceId -> consortiumConfiguration.isEmpty() ? Future.succeededFuture(instanceId)
-              : shareInstanceAmongTenantsIfNeeded(instanceId, consortiumConfiguration.get(), compPOL.getLocations(), requestContext));
-        }
+        Future<String> instanceIdFuture = compPOL.getInstanceId() != null ?
+          Future.succeededFuture(compPOL.getInstanceId()) :
+          getInstanceRecord(compPOL, isInstanceMatchingDisabled, requestContext);
+        consortiumConfiguration.ifPresent(
```

Comment:
return statement should be used, because without it this code share instance functionality will be executed separately and if this call will fail we would not know about an error, our final future will be successful

Reply:
Actually it is returned at line 353. I will replace `?` with `if` to improve readability